utils/fragment_range.hh

```diff
@@ -361,7 +362,7 @@ T read_simple(View& v) {
     if (v.current_fragment().size() >= sizeof(T)) [[likely]] {
         auto p = v.current_fragment().data();
         v.remove_prefix(sizeof(T));
-        return net::ntoh(*reinterpret_cast<const net::packed<T>*>(p));
+        return net::ntoh(read_trivial<T>(p));
```

Comment:
I think (but please don't take what I'm saying at face value, and check), that seastar::read_be can replace net::ntoh(read_trivial), no?
But maybe what you are doing here has the advantage that it is clearer?

Reply:
Yes, seastar::read_be is equivalent to net::ntoh(read_trivial), but read_be is only defined for integers. This particular function only reads integers, so we could use seastar::read_be here, but the nearby read_simple_native also reads non-integers (counter_id), so we need plain read_trivial there. We also need the net::ntoh(read_trivial) for cql_binary_frame_v3, which is another non-integer. I figured using net::ntoh(read_trivial) everywhere is more uniform.